rainier-test/src/test/scala/com/stripe/rainier/compute/CholeskyTest.scala

```diff
@@ -0,0 +1,24 @@
+package com.stripe.rainier.compute
+
+import com.stripe.rainier.sampler.RNG
+
+class CholeskyTest extends ComputeTest {
+    val rng = RNG.default
+    def testRandomSolve(): Unit =  {
+        val rank = rng.int(5) + 2
+        val y = Vector.fill(rank){rng.standardNormal}
+        val packed = Vector.fill(Cholesky.triangleNumber(rank))(rng.standardNormal)
+        val realY = y.map(Real(_))
+        val realPacked = packed.map(Real(_))
+        val realX = Cholesky.lowerTriangularSolve(realPacked, realY)
+        val x = realX.collect{case Scalar(d) => d}
+        val newY = Cholesky.lowerTriangularMultiply(packed.toArray, x.toArray)
+        y.zip(newY).foreach{
+            case (yi, newYi) => assertWithinEpsilon(yi, newYi, "")
+        }
+    }
+
+    test("solve") {
+        0.to(10).foreach{_ => testRandomSolve()}
```

Comment:
why not use scalacheck rather than an ad-hoc approach? With scalacheck you get shrinking and printing of an example test case that fails, which we don't have here it seems.

Reply:
see my comment above. I've had bad experiences with scalacheck. And "shrink" doesn't make sense in this context.